### ISSUE TITLE
The `0` option of `max-empty-lines` reports the run at the tail of an embedded block and its fix leaves it, so the warning is swallowed

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#731]
### What minimal example or steps are needed to reproduce the bug?

Under `0`, `max-empty-lines` counts a line break closing the last line of an embedded block and one more in front of the page's closing markup as an empty line, while its fix writes the `raws.after` of such a root with a maximum of one, so the run is reported and never written, and since a fix callback cannot decline, `--fix` reports the warning as fixed.

Measured on the branch of #586 and on its base `b3409da`, which give the same rows, with `@stylistic/max-empty-lines: 0`:

| input | without `--fix` | with `--fix` | the fixed file linted again |
| --- | --- | --- | --- |
| `<style>\na {}\n\n</style>` | `3:1` | unchanged, nothing reported | `3:1` |
| `<div style="color: pink;\n\n"></div>` | `2:1` | unchanged, nothing reported | `2:1` |
| `<style>\na {}\n</style>` | nothing | unchanged | nothing |
| `a {}\n\n` — the same tail of a plain stylesheet | `2:1` | `a {}\n`, nothing reported | nothing |

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": ["@stylistic/stylelint-plugin"],
	"overrides": [
		{
			"files": ["**/*.html"],
			"customSyntax": "postcss-html",
			"rules": {
				"@stylistic/max-empty-lines": 0
			}
		}
	]
}
```

### What did you expect to happen?

The tail of an embedded block is either counted the way the fix treats it or written to the count the option asks for.

### What actually happened?

The check reports the empty line, the fix leaves it, and `--fix` reports nothing.

### Notes

Found by the review of #586. #585 is the same disagreement at the head of the block, under every option; this one stands at its tail and under `0` alone.
